ocean_lib/web3_internal/test/test_contract_handler_load.py

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import os
+
+import pytest
+from ocean_lib.config_provider import ConfigProvider
+from ocean_lib.web3_internal.contract_handler import ContractHandler
+from ocean_lib.web3_internal.web3_provider import Web3Provider
+from web3.exceptions import InvalidAddress
+
+
+def test_load__fail_empty_artifacts_path():
+    """Tests that an empty artifacts path can not be loaded."""
+    ContractHandler.artifacts_path = None
+    with pytest.raises(AssertionError):
+        ContractHandler._load("DTFactory")
+
+
+def test_load__fail_malformed_eth_address():
+    """Tests that an invalid ETH addres makes the Contract unloadable."""
+    with pytest.raises(InvalidAddress):
+        ContractHandler._load("DTFactory", "foo address")
+
+
+def test_load__fail_wrong_eth_address():
+    """Tests that a different ETH address from the Contract makes it unloadable."""
+    random_eth_address = "0x0daA8DBE3f6760990c886F37E39A5696A4a911F0"
+    with pytest.raises(InvalidAddress):
+        ContractHandler._load("DTFactory", random_eth_address)
+
+
+@pytest.mark.skip(reason="postpone until #202 #227 fixed, then revisit in #185")
+def test_load__name_only():
+    """Tests load() from name-only query."""
+    assert "DTFactory" not in ContractHandler._contracts
+    ContractHandler._load("DTFactory")
+    assert "DTFactory" in ContractHandler._contracts
+
+
+def test_load__name_and_address(network, example_config):
+    """Tests load() from (name, address) query."""
+    addresses = ContractHandler.get_contracts_addresses(
+        network, example_config.address_file
+    )
+    target_address = addresses["DTFactory"]
+
+    test_tuple = ("DTFactory", target_address)
+
+    assert test_tuple not in ContractHandler._contracts
+
+    ContractHandler._load("DTFactory", target_address)
+
+    assert test_tuple in ContractHandler._contracts
+
+
+@pytest.mark.skip(reason="postpone until #202 #227 fixed, then revisit in #185")
+@pytest.mark.nosetup_all  # disable call to conftest.py::setup_all()
+def test_issue185_unit(monkeypatch):
+    """For #185, unit-test the root cause method, which is load"""
+    setup_issue_185(monkeypatch)
+
+    # ensure that conftest::setup_all() was not called
+    assert ConfigProvider._config is None
+    assert Web3Provider._web3 is None
+    assert ContractHandler._contracts == dict()
+    assert ContractHandler.artifacts_path is None
+
+    # actual test. Imports only come now, to avoid setting class-level attributes
+    from ocean_lib.ocean.ocean import Ocean
```

Comment:
there has to be a workaround for this. isort automatically moves these to the top of the file and we'll have pre-commit issues. The simplest would be to set the isort skip flag, but I think this design is not that intuitive anyway. Maybe there is a way to reimport/refresh an imported class?

Reply:
Personal opinion on isort: it causes more trouble than it's worth, everywhere. Eg we can't group imports according to if 3rd party vs our own. We should strongly consider removing it. (I still like black and _some_ code style checker like codacy. But isort, no.)

> Maybe there is a way to reimport/refresh an imported class?

I looked around. It's all super messy.

But we have a specific answer for our purposes: #202 "Web3 provider as singleton is surprising, non-standard, and won't work for multi-chain". This will solve it definitively. And the code to test issue185 will either be simpler in setup, or get deprecated entirely. Hence why _this_ PR is not a full solution to #185, just a partial one to ratchet up quality; and I'm going for #202 next (and #83).

Summary of recommendation here: ignore